### PR TITLE
[HELM-11] added PVC for logs

### PR DESCRIPTION
**Pull request by veronikaKochugova**

### CODE REVIEW

File: mongoose-service/templates/pvc.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  name: {{ .Chart.Name }}-logs-pvc
+spec:
+  storageClassName: {{ .Values.logs.storageClassName }}
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 3Gi
+
+  {{- end }}
```
Comment (dadlex) [inline mongoose-service/templates/pvc.yaml]: I wouldn't mind a newline character here:)

File: mongoose/templates/pvc.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  name:{{ .Chart.Name }}-logs-pvc
+spec:
+  storageClassName: {{ .Values.logs.storageClassName }}
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 3Gi
+
+{{- end }}
```
Comment (dadlex) [inline mongoose/templates/pvc.yaml]: And here

File: mongoose3/templates/controller-pod.yaml [hunk 3]
```diff
@@ -14,6 +14,16 @@ spec:
     configMap:
       name: {{ .Values.name }}-scenario-cm
 
+  {{- if .Values.logs.storageClassName }}
+  - name: logs-volume
+    persistentVolumeClaim:
+      claimName: {{ .Values.name }}-pvc
+  {{- end }}
+
+#  - name: result-volume
```
Comment (dadlex) [inline mongoose3/templates/controller-pod.yaml]: Is it intentional? For later possible use?

File: mongoose3/templates/controller-pod.yaml [hunk 4]
```diff
@@ -37,23 +47,27 @@ spec:
...
                 {{- printf ",\"--storage-auth-uid=%s\"" $.Values.storage.auth.uid -}}
             {{ end }}
 
-            {{ if (ne $.Values.storage.auth.secret "") and (ne $.Values.storage.auth.secret nil) }}
+            {{ if $.Values.storage.auth.secret }}
                 {{- printf ",\"--storage-auth-secret=%s\"" $.Values.storage.auth.secret -}}
             {{ end }}
 
-            {{ if (ne (quote $.Values.storage.net.node.port) "") and (ne $.Values.storage.net.node.port nil) }}
+            {{ if $.Values.storage.net.node.port }}
```
Comment (dadlex) [inline mongoose3/templates/controller-pod.yaml:58]: Does this check for both nil and empty?

File: mongoose3/templates/pvc.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  name: {{ .Chart.Name }}-pvc
+spec:
+  storageClassName: {{ .Values.logs.storageClassName }}
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 3Gi
+
+{{- end }}
```
Comment (dadlex) [inline mongoose3/templates/pvc.yaml]: newline can be added here

**Review by dadlex — APPROVED** [review]

File: mongoose3/templates/controller-pod.yaml (hunk 4 above)
Comment (veronikaKochugova) [reply mongoose3/templates/controller-pod.yaml:58]: Yes

**Pull request merged**